### PR TITLE
Improve transpose sequence

### LINKED ISSUE #1365: GPU implementation of transpose_sequence is too slow
As @jnory mentioned in #1174, `transpose_sequence` is too slow. We need to make a single kernel to make a transposed matrix.

### PR DESCRIPTION
**Pull request by unnonouno**
This patch is about 10x faster than the original. fix #1365

### CODE REVIEW

**Comment by unnonouno** [comment]
This is a test script https://gist.github.com/unnonouno/a53031262b80a9d506c2c3c8bef8d7ed

File: chainer/functions/array/transpose_sequence.py [hunk 1]
```diff
@@ -6,17 +6,55 @@
...
+        offsets1[0] = 0
+        numpy.cumsum([len(x) for x in xs], out=offsets1[1:])
+
+        offsets2 = numpy.empty(length + 1, dtype='i')
+        offsets2[0] = 0
+        numpy.cumsum(lengths, dtype='i', out=offsets2[1:])
+
+        x = xp.concatenate(xs, axis=0)
+        o = xp.empty_like(x)
+        unit = numpy.prod(xs[0].shape[1:], dtype='i')
```
Comment (okuta) [inline chainer/functions/array/transpose_sequence.py]: `xs[0].size // len(xs[0])` is fast.

**Comment by okuta** [comment]
LGTM except comment!

**Comment by unnonouno** [comment]
I fixed it.

**Comment by okuta** [comment]
OK

**Pull request merged**